test/fakeintake/client/client.go

```diff
@@ -240,3 +240,28 @@ func (c *Client) GetCheckRun(name string) ([]*aggregator.CheckRun, error) {
 	}
 	return c.checkRunAggregator.GetPayloadsByName(name), nil
 }
+
+// FlushPayloads sends a request to delete any stored payload
+// and resets client's  aggregators
+func (c *Client) FlushPayloads() error {
```

Comment:
Shouldn't `FlushPayloads` return the existing payloads?
Consider the following example where the client get metrics and then flush payloads:
1. Client get metrics
2. A new metric is received by the fake intake
3. The client call `FlushPayloads()`
In this case the metric received at step `2.` is lost. Did I misunderstand the usage?

Reply:
The initial design was indeed to get values from the api and then reset the server storage, but I considered it would be used to reset the server, discarding payloads received in the meantime. I implemented something that resets at the same time both server and client storage, but as we reset local storage every time to do a get, I could parse payloads here.

I wonder what is more expected, if we should reset everything, if the use case is "reset between tests", or flushing, as reset server but give me all payloads one more time. 

Is that metric at point 2 interesting ?

Reply:
Will change name to `FlushServerAndResetPayloads` and add to description that it should be called to clean up the fakeintake storage in between tests.